plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -0,0 +1,266 @@
+            for (CompletableFuture<GetColumnStatisticsForPartitionResult> future : getStatsFutures.build()) {
+                GetColumnStatisticsForPartitionResult partitionColumnStats = getFutureValue(future, TrinoException.class);
+                for (ColumnStatistics columnStatistics : partitionColumnStats.getColumnStatisticsList()) {
+                    columnStatsMapBuilder.put(
+                            columnStatistics.getColumnName(),
+                            fromGlueColumnStatistics(columnStatistics.getStatisticsData(), tableStatistics.getRowCount()));
+                }
+            }
+            return columnStatsMapBuilder.build();
+        }
+        catch (AmazonServiceException ex) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, ex);
+        }
+    }
+
+    @Override
+    public void updateTableColumnStatistics(Table table, Map<String, HiveColumnStatistics> updatedTableColumnStatistics)
+    {
+        try {
+            HiveBasicStatistics tableStats = getHiveBasicStatistics(table.getParameters());
+            List<ColumnStatistics> columnStats = toGlueColumnStatistics(table, updatedTableColumnStatistics, tableStats.getRowCount());
+            List<List<ColumnStatistics>> columnChunks = Lists.partition(columnStats, GLUE_COLUMN_WRITE_STAT_PAGE_SIZE);
+
+            List<CompletableFuture<Void>> writeChunkFutures = columnChunks.stream().map(columnChunk -> runAsync(
+                    () -> glueClient.updateColumnStatisticsForTable(
+                            new UpdateColumnStatisticsForTableRequest()
+                                    .withCatalogId(catalogId)
+                                    .withDatabaseName(table.getDatabaseName())
+                                    .withTableName(table.getTableName())
+                                    .withColumnStatisticsList(columnChunk))))
+                    .collect(toUnmodifiableList());
+
+            getFutureValue(allOf(writeChunkFutures.toArray(CompletableFuture[]::new)));
+
+            final Map<String, HiveColumnStatistics> currentTableColumnStatistics = this.getTableColumnStatistics(table);
+            Set<String> removedStatistics = difference(currentTableColumnStatistics.keySet(), updatedTableColumnStatistics.keySet());
+            List<CompletableFuture<Void>> writeFutures = removedStatistics.stream()
+                    .map(column -> runAsync(() ->
+                            glueClient.deleteColumnStatisticsForTable(
+                                    new DeleteColumnStatisticsForTableRequest()
+                                            .withCatalogId(catalogId)
+                                            .withDatabaseName(table.getDatabaseName())
+                                            .withTableName(table.getTableName())
+                                            .withColumnName(column)))
+                    ).collect(toUnmodifiableList());
```

Comment:
nit: move closing brackets to previous line. I wonder why checkstyle didn't complain though.